diff --git a/Config.cmake.in b/Config.cmake.in
index 20416a12..dc844a7e 100644
--- a/Config.cmake.in
+++ b/Config.cmake.in
@@ -9,7 +9,7 @@ foreach(component ${phasar_FIND_COMPONENTS})
 endforeach()

 function(phasar_config executable)
-  find_package(LLVM 13 REQUIRED CONFIG)
+  find_package(LLVM 12 REQUIRED CONFIG)
   include_directories(${LLVM_INCLUDE_DIRS})
   link_directories(${LLVM_LIB_PATH} ${LLVM_LIBRARY_DIRS})
   find_library(LLVM_LIBRARY NAMES LLVM HINTS ${LLVM_LIBRARY_DIRS})

