diff --git a/crt/crt1.c b/crt/crt1.c
index 8fe8ab5d..26cbfa04 100644
--- a/crt/crt1.c
+++ b/crt/crt1.c
@@ -1,4 +1,6 @@
+#include <elf.h>
 #include <features.h>
+#include <stddef.h>
 #include "libc.h"
 
 #define START "_start"
@@ -15,5 +17,12 @@ void _start_c(long *p)
 {
 	int argc = p[0];
 	char **argv = (void *)(p+1);
+
+	char **envp = argv+argc+1;
+	int i;
+	for (i=0; envp[i]; i++);
+	size_t *auxv = (void *)(envp+i+1);
+	for (i=0; auxv[i]; i+=2) if (auxv[i]==AT_RANDOM) auxv[i+1] = 0;
+
 	__libc_start_main(main, argc, argv, _init, _fini, 0);
 }
