From fc2be4828b755c730be9955b4fec9247cd386677 Mon Sep 17 00:00:00 2001
Date: Thu, 17 Nov 2022 16:46:09 -0600
Subject: [PATCH] Disable putting the optnone attribute on functions compiled
 with -O0

---
 shared/compiler.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/shared/compiler.go b/shared/compiler.go
index 8192583..8f79d81 100644
--- a/shared/compiler.go
+++ b/shared/compiler.go
@@ -286,7 +286,7 @@ func buildBitcodeFile(compilerExecName string, pr ParserResult, srcFile string,
 	args := pr.CompileArgs[:]
 	//iam: 03/24/2020 extend with the LLVM_BITCODE_GENERATION_FLAGS if any.
 	args = append(args, LLVMbcGen...)
-	args = append(args, "-emit-llvm", "-c", srcFile, "-o", bcFile)
+	args = append(args, "-emit-llvm", "-Xclang", "-disable-O0-optnone", "-c", srcFile, "-o", bcFile)
 	success, err := execCmd(compilerExecName, args, "")
 	if !success {
 		LogError("Failed to build bitcode file for %s because: %v\n", srcFile, err)
-- 
2.17.1

